app/models/scheduled_database_check_job.rb

```diff
@@ -0,0 +1,18 @@
+class ScheduledDatabaseCheckJob
+  def self.perform(job_type)
+    case job_type
+    when 'clean_subscription_series'
+      Subscription.where(subscribable_type: "Series").includes(:subscribable).find_each do |s| 
+        s.delete if s.name.nil? && Series.find_by_id(s.subscribable_id).nil?
```

Comment:
All righty. If we ever end up with stuff hanging around, we can reconsider our paranoia, but this should be good :+1: